dashboard/app/views/levels/_level_group.haml

```diff
@@ -6,21 +6,47 @@
 
 = render partial: 'levels/common_audio'
 
-- data["levels"].each_with_index do |level_name, index|
+- if data["per_page"]
+  - # use a 0-based page internally, though the URL is 1-based.
+  - page = params[:page] ? (params[:page].to_i - 1) : 0
+  - per_page = data["per_page"].to_i
+  - first_level = page * per_page
+  - last_level = [first_level + per_page - 1, data["levels"].length - 1].min
+- else
+  - first_level = 0
+  - last_level = data["levels"].length - 1
+
+- data["levels"][first_level..last_level].each_with_index do |level_name, index|
   - level_blob = Level.find_by_name(level_name)
   - level = level_blob
   - # Parse the array of the level resuts; see sample below in JavaScript.
   - if @last_attempt
-    - last_attempt = JSON.parse(@last_attempt).find{|item| item["level_id"]==level.id}["result"]
+    - last_attempt_entry = JSON.parse(@last_attempt).find{|item| item["level_id"]==level.id}
+    - last_attempt = last_attempt_entry["result"] if last_attempt_entry
   - else
     - last_attempt = ""
-  = render partial: 'levels/single_multi', locals: {index: index, standalone: false, level: level, last_attempt: last_attempt }
+  = render partial: 'levels/single_multi', locals: {page: page, index: index, standalone: false, level: level, last_attempt: last_attempt }
 
 = render partial: 'levels/common_audio'
-= render partial: 'levels/dialog', locals: {app: app, data: data}
+
+- if data["per_page"] && last_level < data["levels"].length - 1
+  %a.btn.btn-large.btn-primary.nextPageButton
+    =t('nextPage')
+- else
+  = render partial: 'levels/dialog', locals: {app: app, data: data}
+  %a.btn.btn-large.btn-primary.unsubmitButton{style: "display: none"}
+    =t('unsubmit')
 
 :javascript
   #{render template: 'levels/multi.js', formats: [:js]}
 
-  var levelCount = #{data["levels"].count};
+  var levelCount = #{last_level - first_level + 1};
+  var page = #{page || -1};
+
+  var fallbackResponse = #{@fallback_response.to_json};
```

Comment:
Oh, you get these in appOptions if you include the levels/dialog partial which in turn uses _dialogHelper.js which includes a bunch of code for showing dialogs and submitting results.  In non-final pages of a multi-page LevelGroup, it seemed much simpler to do a sendReport using these four values and leave out that code.